### PR TITLE
Add test for digest auth

### PR DESCRIPTION
**Pull request by bastien-phi**

### CODE REVIEW

File: tests/server/public/index.php [hunk 1]
```diff
@@ -58,13 +58,45 @@ function build_response($request)
...
+});
+
 $app->post('/multi-part', function () {
     return response()->json([
-        'body_content' => app('request')->only(['foo', 'baz']),
-        'has_file' => app('request')->hasFile('test-file'),
-        'file_content' => file_get_contents($_FILES['test-file']['tmp_name']),
-        'headers' => app('request')->header(),
-    ], 200);
+                                'body_content' => app('request')->only(['foo', 'baz']),
```
Comment (sixlive) [inline tests/server/public/index.php]: Looks like the spacing got thrown off here by accident.

File: tests/server/public/index.php [hunk 2]
```diff
@@ -58,13 +58,45 @@ function build_response($request)
...
+    if ($data['username'] != 'zttp') {
+        return response(null, 401);
+    }
+
+    $a = md5('zttp:' . $realm . ':secret');
+    $b = md5(app('request')->server->get('REQUEST_METHOD') . ':' . $data['uri']);
+    $validResponse = md5($a . ':' . $data['nonce'] . ':' . $data['nc'] . ':'.$data['cnonce'] . ':' . $data['qop'] . ':' . $b);
+
+    if ($data['response'] != $validResponse) {
+        return response(null,401);
```
Comment (sixlive) [inline tests/server/public/index.php]: Should add some space to the arguments `response(null, 401)`

**Comment by adamwathan** [comment]
Will merge this if we can fix conflicts. Still blows my mind how much code has to be written by hand to check this shit 😱

**Comment by bastien-phi** [comment]
I've clearly not written all of that by hand. Copy-paste most of it from http://php.net/manual/en/features.http-auth.php

**Comment by adamwathan** [comment]
@bastien-phi Haha yeah sorry what I really just meant is that I can't believe there's not built in functions for handling it.

Merging now, thanks!

**Pull request merged**